### PR TITLE
Distributed AnnDataCollection

### PR DESCRIPTION
**Pull request by ordabayevy**

### CODE REVIEW

File: test/test_distributed_anndata.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+)
+def test_indexing(adt, dat, select):
+    # compare indexing single and distributed anndata
+    dat_view = dat[select, :2]
+    adt_view = adt[select, :2]
+    np.testing.assert_array_equal(adt_view.X, dat_view.X)
+    np.testing.assert_array_equal(adt_view.var_names, dat_view.var_names)
+    np.testing.assert_array_equal(adt_view.layers["L"], dat_view.layers["L"])
+    np.testing.assert_array_equal(adt_view.obsm["M"], dat_view.obsm["M"])
+    np.testing.assert_array_equal(dat_view.obs["A"], dat_view.obs["A"])
```
Comment (mbabadi) [inline test/test_distributed_anndata.py:93]: Additional tests:
- different max cache sizes (1, 2, 3)
- failure in strict mode if __get_item__ indices span too many chunks
- success in non-strict mode if __get_item__ indices span too many chunks
- explicit asserts that no more than maxsize chunks are in the LRU cache
- additional gene-level slicing than `, :2]` in all tests

**Review by mbabadi — CHANGES_REQUESTED** [review]
Back to you @ordabayevy :)

**Comment by ordabayevy** [comment]
Back to you @mbabadi :) Address all the comments.

**Comment by ordabayevy** [comment]
Removed `expand_urls` altogether since we only need simple `braceexpand`. There might be inefficiencies in `_normalize_indices` and `_resolve_idx` when the number of files is too big but I decided to address it in another issue or PR.

**Comment by mbabadi** [comment]
@ordabayevy great! please make an issue about potential inefficiencies in `_normalize_indices` and `_resolve_idx` and feel free to merge!

**Review by mbabadi — APPROVED** [review]

**Comment by ordabayevy** [comment]
Thanks for reviewing @mbabadi !

**Pull request merged**